recipes/mastercard/runner.sh

```diff
@@ -59,52 +57,62 @@ AWS_DEFAULT_REGION=us-east-1
-    # Then use awk to select filename.
-    unzip -d /input -P $MASTERCARD_PASSWORD $FULL_FILENAME || exit 519
-    #removes all downloaded non-csv files.
-    rm $(find $BASEDIR/input -type f -not -name "*.csv")
-    #cd $BASEDIR
-
-    #find the filename. Should only be one file. 
-    KEY=$(ls /input)
-    FILENAME=$(basename $KEY)
-    echo $FILENAME
-
-    #send csv to PSQL
-    cat /input/$FILENAME | psql $RDP_DATA -v NAME=$NAME -v VERSION=$VERSION -f create_mastercard.sql
-    #clean up
-    rm -rf input
-    (
-        cd output
+    #loop through all the files and add each to output
+    echo 'listing...'
+    MYFILES=$(ls input | grep .zip)
+    echo "MYFILES:" $MYFILES
+    mkdir -p output
+    for FULL_FILENAME in $MYFILES
+        do 
+        #loop begins
+        
+        FILENAME=${FULL_FILENAME%.*}
+
+
+        pushd input
+        rm *.csv || echo "Failed to remove any csvs"
+        popd
+        echo "unzipping " $FULL_FILENAME
+        unzip -d ./input -P $MASTERCARD_PASSWORD ./input/$FULL_FILENAME || exit 519
+
+  
+        #find the csv. There should only be one.
+        pushd input
+        CSV_FILENAME=$(ls *.csv)
+        #CSV_FILENAME=$(ls | egrep '\.csv')
+        popd
+        #send csv to PSQL
+        cat ./input/$CSV_FILENAME | psql $RDP_DATA -v NAME=$NAME -v VERSION=$VERSION -f create_mastercard.sql
+        
+
+        (
         psql $RDP_DATA -c "\COPY (
             SELECT * FROM $NAME.\"$VERSION\"
-            ) TO stdout DELIMITER ',' CSV HEADER;" > mastercard.csv
+            ) TO stdout DELIMITER ',' CSV HEADER;" > ./output/mastercard_$FILENAME.csv
         
         #Write Version info
-        echo "$VERSION" > version.txt
-    )
+        echo "version: " $VERSION
+        echo "$VERSION_$FILENAME" >> ./output/version.txt
+        )
     
-    #unsplit csv is too large. Must compress.
-    echo "compressing mastercard.csv"
-    zip -9 output/daily_transactions.zip output/mastercard.csv
+        #unsplit csv is too large. Must compress.
+        echo "compressing mastercard_$FILENAME.csv"
+        zip -9 ./output/daily_transactions_$FILENAME.zip output/mastercard_$FILENAME.csv
     
-    #If you don't remove unsplit csv, sharepoint.py will overflow the RAM and the process killed when it tries to upload it.
-    rm -rf output/mastercard.csv
-
-    #Upload uploads everything in the output folder.
-    Upload $NAME $VERSION
-    Upload $NAME latest
-    rm -rf output
-    Version $NAME '' $VERSION $NAME
-    if [ "$AWS_ERROR" -eq 1 ]
-    then
-        echo "Sharepoint upload successfull but failed to upload to AWS.";
-        exit 435;
-    fi
+        #If you don't remove unsplit csv, sharepoint.py will overflow the RAM and the process killed when it tries to upload it.
+        rm -rf output/mastercard_$FILENAME.csv       
+    done
+    #loop ends
+
+        #Upload uploads everything in the output folder.
+        Upload $NAME $VERSION
```

Comment:
Should lines 108-111 be part of the loop above? If not, do you need to remove the indent?